### DISCUSSION TITLE
Stream audiobook from an HTTP server

### DISCUSSION THREAD

**Discussion by codingCoffee** [PaulWoitaschek/Voice#1324] [Ideas]
**Is your feature request related to a problem? Please describe.**
It is more of a feature request. I'd like to access (stream/listen to) audiobooks hosted on a server, as opposed to from local storage. 

**Describe the solution you'd like**
Just as we have an option to add a local folder as a location, I'd like another option to add a URL to a server. Something along the lines of "https://username:password@sample.site/path/to/audiobooks". This would be a simple HTTP/s server which hosts audiobooks. Upon adding this path, Voice should cache the necessary metadata about the available books present on the server locally, and when an audiobook is clicked, it should be streamed from the server. Maybe an option to download the audiobook is also something which can be considered. 

**Describe alternatives you've considered**
I haven't found any viable alternatives as such. However, something like [airsonic](https://github.com/airsonic-advanced/airsonic-advanced) comes kinda close, as it allows you to stream music from a server, as opposed to local storage. But then again this is not an android app, and it is more catered towards music and not audiobooks. Currently, I go about downloading the audiobook manually and placing them in the folder of my choice. Having a way to stream could greatly enhance the experience. 

**Additional context**
I do realize this is not a feature set a lot of people might want, but I felt like mentioning it just the same. And THANK YOU so much for making such a brilliant app. I have not found any other android app which comes even close to Voice. Be it in terms of the simplicity of its UI, or its feature set, which although minimal has all the required features one wants like the ability to modify playback speed, skip silence, etc.

**Comment by thw0rted** [comment]
There are several requests for integration with e.g. Google Drive and Plex, but this is the first I've seen for simple http. I'd use it, if it were added.
